### DISCUSSION TITLE
Undefined constant "Yajra\Pdo\OCI_DEFAULT"

### DISCUSSION THREAD

**Discussion by aurawindsurfing** [yajra/laravel-oci8#791] [Ideas]
Hi @yajra 

I have come across that issue and searched issued on github. What I found the problem was for people saying - it does not work on production is that on production you need to configure extension in 2 places in CLI and FPM. When they try tro troubleshoot from CLI everything looks ok but still does not work ;-)

Maybe would be good idea for your package to actually throw and error saying that oci8 is not installed and suggesting those 2 places to check?

Just an idea.

Thanks a lot for a great package!

**Comment by yajra** [comment]
`ext-oci8` is already a requirements via [composer.json](https://github.com/yajra/laravel-oci8/blob/10.x/composer.json#L19). However, as you've mentioned it needs to be configured on CLI and FPM on some instances. Composer is on CLI level which most of the time works. Setting up on FPM sometimes requires special configurations. 

I also still struggle sometimes on setting this up since setup keeps on changing. Just like on the package tests now, it's failing because of `oci8` installation ^_^.

Anyways, thanks for the inputs. I will try to check if we can do something to improve the error handling.

**Comment by aurawindsurfing** [comment]
Hey! most people will probably use Laravel Forge and there are 2 configs you can edit in PHP configuration. Which is exactly the issue here. would be worth mentioning maybe in the installation part or something. But yes installation could be the hardest part of using your package since after that it works like a charm ;-)

Have a great day!

**Reply by yajra** [reply]
TBH, I haven't used forge for Oracle. It would be great if you can PR your findings when using Forge maybe in readme? Thanks in advance and for the sponsorship 🍻

**Reply by aurawindsurfing** [reply]
Will do!
